cocos/base/ObjectFactory.h

```diff
@@ -44,12 +44,9 @@ class CC_DLL ObjectFactory
         TInfo();
         TInfo(const std::string& type, Instance ins = nullptr);
         TInfo(const std::string& type, const InstanceFunc& ins = nullptr);
-        TInfo(const TInfo &t);
-        ~TInfo();
-        TInfo& operator= (const TInfo &t);
         std::string _class;
-        Instance _fun;
-        InstanceFunc _func;
+        Instance _fun = nullptr;
```

Comment:
Why this class has both c pointer function and std::function? std::function should be enough.

Reply:
`_fun` is used in lua bindings. I am not sure how it works.

 https://github.com/cocos2d/cocos2d-x/blob/4107942cbf0b674883d1db325800c50d0d890240/cocos/scripting/lua-bindings/script/cocostudio/CocoStudio.lua#L30-L44

Reply:
I think that link cocos2d-x/cocos/scripting/lua-bindings/script/cocostudio/CocoStudio.lua is just the same implementation but in Lua.
So we can remove _fun from C++ and have only std::function